evennia/contrib/game_systems/storage/storage.py

```diff
@@ -0,0 +1,180 @@
+from evennia import CmdSet
+from evennia.utils import list_to_string
+from evennia.utils.search import search_object_by_tag
+from evennia.commands.default.muxcommand import MuxCommand
+
+
+class StorageCommand(MuxCommand):
+    """
+    Shared functionality for storage-related commands
+    """
+
+    def at_pre_cmd(self):
+        """
+        Check if the current location is tagged as a storage location
+        Every stored object is tagged on storage, and untagged on retrieval
+        """
+        success = super().at_pre_cmd()
+        self.storage_location_id = self.caller.location.tags.get(category="storage_location")
+        if not self.storage_location_id:
+            self.caller.msg(f"You cannot {self.cmdstring} anything here.")
+            return True
+
+        self.object_tag = (
+            "shared" if self.storage_location_id.startswith("shared_") else self.caller.pk
```

Comment:
Optional style comment: Can be a little cleaner to add a module constant like `TAG_NAME` at the top and use that here rather than inputing the string explicitly where it's used.